src/monsters.cpp

```diff
@@ -767,9 +715,25 @@ bool Monsters::loadMonster(const std::string& file, const std::string& monster_n
 		return false;
 	}
 
-	if (mType == nullptr) {
-		mType = new MonsterType();
+	if (reloading) {
+		uint32_t id = getIdByName(monster_name);
```

Comment:
Try not to overshadow other variables, this overshadows the static id. You could just move the static id down a bit and keep the names.